Loop/Managers/LoopDataManager.swift

```diff
@@ -397,13 +397,9 @@ final class LoopDataManager: ObservableObject {
         }
 
         var overrides = temporaryPresetsManager.overrideHistory.getOverrideHistory(startDate: neededSensitivityTimeline.start, endDate: forecastEndTime)
-
-        // Bug (https://tidepool.atlassian.net/browse/LOOP-4759) pre-meal is not recorded in override history
-        // So currently we handle automatic forecast by manually adding it in, and when meal bolusing, we do not do this.
-        // Eventually, when pre-meal is stored in override history, during meal bolusing we should scan for it and adjust the end time
-        if !disablingPreMeal, let preMeal = temporaryPresetsManager.preMealOverride {
-            overrides.append(preMeal)
-            overrides.sort { $0.startDate < $1.startDate }
+        
+        if disablingPreMeal, let preMealOverride = temporaryPresetsManager.preMealOverride {
+            overrides.remove(preMealOverride)
```

Comment:
It might be better conceptually to mark the end time rather than removing it entirely from the list. Right now it won't make any functional difference, since meal overrides do not adjust sensitivity, and target range adjustments in the past do not affect the forest.  But if either of those change; if meal presets adjust sensitivity, or the algorithm changes in a way that past targets affect future predictions, then we would need to make sure the historical part of the premeal override was included.

And as I mentioned in dev sync, if you want to make a `getOverrideHistory` method that can return an overrides list with the premeal truncated,  if it's simpler, then that works.

Reply:
It does seem like I've got it working with `.now` as the scheduled end time but I still have a concern about the scheduledEndTime being set to `.now` when it's possible the bolus confirmation, and thus the actual end time of the preset being later once the users confirms.

Reply:
The end time should be the time of the forecast (`baseTime`).

Reply:
Just for the forecast. Remember the forecast for a recommendation is just a "what if" type of thing.

Reply:
And we don't actually end the preset; we're just modifying the data that is used for forecast generation.  The actual Loop state is that the preset is still running.